packages/block-editor/src/components/block-preview/auto.js

```diff
@@ -55,7 +55,7 @@ function ScaledBlockPreview( {
 	}, [ styles, additionalStyles ] );
 
 	// Initialize on render instead of module top level, to avoid circular dependency issues.
-	MemoizedBlockList = MemoizedBlockList || pure( BlockList );
+	MemoizedBlockList = MemoizedBlockList || memo( BlockList );
```

Comment:
Hm, this feels super weird! Does this even work? :D

Reply:
I think so 😅 there is even a Babel plugin that applies similar optimizations - https://babeljs.io/docs/babel-plugin-transform-react-constant-elements.html

Reply:
This was marked as an `error` by the new `eslint-plugin-react-compiler` plugin - https://github.com/WordPress/gutenberg/actions/runs/9158471368/job/25176908344?pr=61788#step:5:86.